api/core/rag/datasource/vdb/tidb_vector/tidb_vector.py

```diff
@@ -65,7 +65,7 @@ def __init__(self, collection_name: str, config: TiDBVectorConfig, distance_func
         super().__init__(collection_name)
         self._client_config = config
         self._url = (f"mysql+pymysql://{config.user}:{config.password}@{config.host}:{config.port}/{config.database}?"
-                     f"ssl_verify_cert=true&ssl_verify_identity=true")
+                     f"ssl_verify_cert=true&ssl_verify_identity=true&program_name=langgenius/dify")
```

Comment:
Please consider moving this into ENV vars. Try not to hard coded in the program.

Reply:
The purpose of this attribute is to identify the application is from dify. I think there is no benefit to change it and the extra configuration exposes unnecessary detail to the end user. Maybe make it a constant in root module can be an alternative. And it can be used in other integrations if they have something similar to MySQL's connection attributes. 

What do you think?

Reply:
Constant looks good to me.

Reply:
> Constant looks good to me.

Hello @crazywoola
It's in config.py as a constant like `CURRENT_VERSION` now. Please take an other look, thanks.

Reply:
Hm, just realized config.py has been refactored and there is no CURRENT_VERSION in it now.

Reply:
Since the `config.py` has been refactored, and those ENV vars are loaded from `.env` file, I guess it's more appropriate to add this into `.env.example` and `.env`.

We are also working on https://github.com/langgenius/dify/pull/4299/files#diff-ec10582dde9af6c0c622084d3244b29d496260613491c430ec8958565e2f31b4 which changed a lot in those configurations.

Reply:
But I am ok with this, maybe we can open another pr later to adjust things.